enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,413 @@
+
+### User Stories
+
+As a developer using the Shared Resource CSI Driver
+I want validation for `readOnly: true` to be set on volumes for the Shared Resource CSI Driver to occur prior
+to the Pod Spec'able object getting persisted to etcd, so that pods are not stuck in "Creating" state waiting for a mount that will never succeed.
+Rather, I learn of the problem on the API Object PUT/POST.
+
+As an OpenShift operator maintainer or cluster administrator
+I want to reserve the `openshift-` prefix for SharedSecrets and SharedConfigMaps
+so that future OpenShift operators can create system-level shared resources and not have name conflicts 
+with items created by OCP users.
+
+As a developer, OpenShift operator maintainer, or cluster administrator
+I want options that verify the Secret or ConfigMap content in a SharedSecret or SharedConfigMap is "valid" for 
+any of the "well known" SharedSecrets or SharedConfigMaps that OCP could pre-populate the cluster with (entitlement
+secret for example).
+
+As a developer using the Shared Resource CSI Driver I want any validations to be provisioned and lifecycled in as much
+as possible the same fashion as the driver itself for consistency (i.e. via Cluster Storage Operator).
+
+As a developer or administrator using the Shared Resource CSI Driver I want sufficient metrics on the validations' prevention of incorrect Pods,
+SharedConfigMaps, and SharedSecrets, as a means of understanding the need for correcting misuse..
+
+As a developer or administrator using the Shared Resource CSI Driver I want necessary debug of the validation mechanism to be captured by
+OCP must-gather process.
+
+
+### API Extensions
+
+Quick note:  When discussing API, 'Shared Resource' on its own == SharedConfigMap / SharedSecret, and 'Reference' == the corresponding
+Secret or ConfigMap.
+
+A [ValidatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook)
+will be used to inspect all created/modified Shared Resources, and make sure their names do not start
+with the reserved prefix "openshift-" unless their Shared Resource name and Reference key are found in a 
+well known list of OCP pre-populated Shared Resource.  That list will be in well known ConfigMaps that resides 
+in the managed CSI Driver namespace.  Components / operators other than Shared Resources that want to create pre-populated
+SharedSecrets or SharedConfigMaps should submit PRs against the Shared Resource CSI Driver Operator Repo to update the 
+corresponding ConfigMap with the necessary information.  This is akin to how we create in-payload image streams via the PRs against the 
+samples operator today.
+
+There will be a "ocp-sharedsecret-list" and "ocp-sharedconfigmap-list" ConfigMap.  The key of an entry is the Shared Resource
+name.  The value should be of the form "namespace:name" in order to capture. 
```

Comment:
May help to provide an example here with the Insights operator's cluster entitlement:

```
kind: ConfigMap
apiVersion: v1
metadata:
  name: ocp-sharedsecret-list
  namespace: <namespace of the webhook>
data:
  openshift-etc-pki-entitlement: openshift-config-managed:etc-pki-entitlement
```

Reply:
will do in next push